### PR TITLE
Validate method names in InvokeInstruction

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java [hunk 1]
```diff
@@ -244,13 +247,14 @@ else if (type == Void.class) {
...
     {
+        checkIdentifier(name);
         return new Block(context)
                 .invokeDynamic(name, binding.getType(), binding.getBindingId());
     }
 
     public static ByteCodeNode invoke(CompilerContext context, Binding binding, Signature signature)
     {
-        return invoke(context, binding, signature.getName());
+        return invoke(context, binding, mungeIdentifier(signature.getName()));
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java]: I think this code should verify, but the caller should do the munging.

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java]: This the method that takes a Presto `Signature` which can contain invalid Java names. The other method that takes a name only does verification.

File: presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java [hunk 2]
```diff
@@ -290,4 +294,27 @@ public static ByteCodeNode generateWrite(CallSiteBinder callSiteBinder, Compiler
                                 .invokeInterface(Type.class, methodName, void.class, BlockBuilder.class, type.getJavaType()))
                         .build());
     }
+
+    public static void checkIdentifier(String name)
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/ByteCodeUtils.java]: add the check to the invoke `ByteCodeNode` so we fail fast

**Comment by dain** [comment]
Looks good

**Comment by dain** [comment]
looks good

**Pull request merged**